### PR TITLE
Add SDK creation and customization guide

### PR DESCRIPTION
**Pull request by ice1000**

### CODE REVIEW

File: reference_guide/project_model/sdk.md [hunk 1]
```diff
@@ -39,3 +39,68 @@ Sdk projectSdk = ProjectRootManager.getInstance(project).getProjectSdk();
   ```
   
 See the following [code sample](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/project_model/src/com/intellij/tutorials/project/model/ProjectSdkAction.java) to get more familiar with SDK manipulation tool set.
+
+## Working with your own SDK
+
+To create your own SDK, You need to create a class extends `SdkType`, and set your recommended SDK name, SDK home path, a presentable name. <br/>
+Finally you need to implement `isValidSdkHome` to do the SDK validation, and this example just check if the given path is a directory.
```
Comment (zolotov) [inline reference_guide/project_model/sdk.md]: I don't see much sense in this. `isValidSdkHome` is an abstract method, it's obvious that user should implement it.

File: reference_guide/project_model/sdk.md [hunk 2]
```diff
@@ -39,3 +39,68 @@ Sdk projectSdk = ProjectRootManager.getInstance(project).getProjectSdk();
   ```
   
 See the following [code sample](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/project_model/src/com/intellij/tutorials/project/model/ProjectSdkAction.java) to get more familiar with SDK manipulation tool set.
+
+## Working with your own SDK
+
+To create your own SDK, You need to create a class extends `SdkType`, and set your recommended SDK name, SDK home path, a presentable name. <br/>
```
Comment (zolotov) [inline reference_guide/project_model/sdk.md]: All these methods are abstract, there is no need to say that they must be implemented.

File: reference_guide/project_model/sdk.md [hunk 3]
```diff
@@ -39,3 +39,68 @@ Sdk projectSdk = ProjectRootManager.getInstance(project).getProjectSdk();
...
+    @Override public boolean setupSdkPaths(@NotNull Sdk sdk, @NotNull SdkModel sdkModel) {
+        SdkModificator modificator = sdk.getSdkModificator();
+        modificator.setVersionString(getVersionString(sdk));
+        modificator.commitChanges();
+        return true;
+    }
+}
+```
+
+Optionally, you can override `getIcon` to set an icon to your SDK type.
```
Comment (zolotov) [inline reference_guide/project_model/sdk.md]: getIcon is self-descriptive

File: reference_guide/project_model/sdk.md [hunk 4]
```diff
@@ -39,3 +39,68 @@ Sdk projectSdk = ProjectRootManager.getInstance(project).getProjectSdk();
...
   
 See the following [code sample](https://github.com/JetBrains/intellij-sdk-docs/blob/master/code_samples/project_model/src/com/intellij/tutorials/project/model/ProjectSdkAction.java) to get more familiar with SDK manipulation tool set.
+
+## Working with your own SDK
+
+To create your own SDK, You need to create a class extends `SdkType`, and set your recommended SDK name, SDK home path, a presentable name. <br/>
+Finally you need to implement `isValidSdkHome` to do the SDK validation, and this example just check if the given path is a directory.
+
+```java
+import com.intellij.openapi.projectRoots.*;
```
Comment (zolotov) [inline reference_guide/project_model/sdk.md]: The methods used in example either self-descriptive or contain javadoc. I'd say it's better to add documentation to `setupSdkPath` and provide a link to `SdkType` class.

File: reference_guide/project_model/sdk.md [hunk 5]
```diff
@@ -39,3 +39,68 @@ Sdk projectSdk = ProjectRootManager.getInstance(project).getProjectSdk();
...
+    }
+}
+```
+
+Optionally, you can override `getIcon` to set an icon to your SDK type.
+
+The last method is not abstract, so it's possible to remove it. But to save your SDK settings
+(say, the next time you open your IDE, the SDK will still be there), you should override it.
+
+To let users create or select an existing SDK, you may do this in some configuration panels (like, module wizard, facet, or run configuration):
```
Comment (zolotov) [inline reference_guide/project_model/sdk.md]: > To let users create or select an existing SDK,

I don't see how ProjectJdksEditor is related to an ability to create SDK.

It's not clear where to put this code. I believe it's enough to provide a link to `ProjectJdksEditor`.

File: reference_guide/project_model/sdk.md (hunk 2 above)
Comment (ice1000) [reply reference_guide/project_model/sdk.md]: Will remove.

File: reference_guide/project_model/sdk.md (hunk 1 above)
Comment (ice1000) [reply reference_guide/project_model/sdk.md]: I'll edit it.

File: reference_guide/project_model/sdk.md (hunk 4 above)
Comment (ice1000) [reply reference_guide/project_model/sdk.md]: Link provided.

File: reference_guide/project_model/sdk.md (hunk 3 above)
Comment (ice1000) [reply reference_guide/project_model/sdk.md]: Removed.

File: reference_guide/project_model/sdk.md (hunk 5 above)
Comment (ice1000) [reply reference_guide/project_model/sdk.md]: Updated.

**Comment by zolotov** [comment]
thanks!

**Pull request merged**